src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupIntrinsicTask.cs

```diff
@@ -552,7 +561,11 @@ private List<ProjectItemInstance> FindItemsMatchingSpecification
                     // Don't unescape wildcards just yet - if there were any escaped, the caller wants to treat them
                     // as literals. Everything else is safe to unescape at this point, since we're only matching
                     // against the file system.
-                    string[] fileList = EngineFileUtilities.GetFileListEscaped(Project.Directory, piece);
+                    (string[] fileList, FileMatcher.SearchAction action) = EngineFileUtilities.GetFileListEscaped(Project.Directory, piece);
+                    if (action == FileMatcher.SearchAction.ReturnLogDriveEnumerationWildcard)
+                    {
+                        LoggingContext.LogWarning($"Drive enumeration was detected during wildcard expansion, for the Exclude attribute, which relied on the file spec {piece}.");
```

Comment:
Test:
```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <Target Name="TestTarget">
        <ItemGroup>
            <FilesToCopy Include="$(Microsoft_WindowsAzure_EngSys)\*.pdb" Exclude="$(Microsoft_WindowsAzure_EngSys)\**\*" />
        </ItemGroup>
    </Target>
</Project>
```

Output:
![image](https://user-images.githubusercontent.com/89041446/144304995-f281fadc-46ec-472b-873e-89f8e9b0b042.png)

Reply:
Yes, with new changes (& more appropriate test) -

Test:
```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <Target Name="TestTarget">
        <ItemGroup>
            <FilesToCopy Include="c:\src\*.log" Exclude="$(Microsoft_WindowsAzure_EngSys)\**\*" />
        </ItemGroup>
    </Target>
</Project>
```
Output:
![image](https://user-images.githubusercontent.com/89041446/144507519-bb10e12d-5d36-4736-bd92-b458ae330f7b.png)

Reply:
Yes, with new changes, the following output appears for both the old and new test --

New Test:
```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <Target Name="TestTarget">
        <ItemGroup>
            <FilesToCopy Include="c:\src\*.log" Exclude="$(Microsoft_WindowsAzure_EngSys)\**\*" />
        </ItemGroup>
    </Target>
</Project>
```
Output:
![image](https://user-images.githubusercontent.com/89041446/144507519-bb10e12d-5d36-4736-bd92-b458ae330f7b.png)

Reply:
Yes, with new changes, the following output appears for both the old and new test --
```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <Target Name="TestTarget">
        <ItemGroup>
            <FilesToCopy Include="c:\src\*.log" Exclude="$(Microsoft_WindowsAzure_EngSys)\**\*" />
        </ItemGroup>
    </Target>
</Project>
```
Output:
![image](https://user-images.githubusercontent.com/89041446/144509514-2e2a09ae-6d2f-4e73-9acb-095d4174435c.png)